src/serde/de.rs

```diff
@@ -1330,7 +1330,7 @@ where
     // check JSON size, because the design of `sonic_rs::Value`, parsing JSON larger than 4 GB is
     // not supported
     let len = read.as_u8_slice().len();
-    if len >= (1 << 32) {
+    if len >= (u32::MAX / 2) as _ {
```

Comment:
should be  > u32::MAX ？ because the max allowed length is 4 Gb - 1

Reply:
Ah, the docs mentioned 2GB

Reply:
At least when I looked? Maybe I had an outdated branch

Reply:
yeah, it should be 4 GB here